core/drivers/crypto/caam/acipher/caam_rsa.c

```diff
@@ -539,18 +546,32 @@ static TEE_Result do_gen_keypair(struct rsa_keypair *key, size_t key_size)
 		goto exit_gen_keypair;
 	}
 
-	/* First allocate primes p and q in one buffer */
-	retstatus = caam_calloc_align_buf(&genkey.p, key_size / 8);
+	size_d_gen_val_ptr = caam_calloc_align(sizeof(uint32_t));
+	if (!size_d_gen_val_ptr) {
+		ret = TEE_ERROR_OUT_OF_MEMORY;
+		goto exit_gen_keypair;
+	}
+
+	/* First allocate primes p and q */
+	genkey.p.key_type = CAAM_KEY_PLAIN_TEXT;
+	genkey.p.sec_size = (key_size / 8) / 2;
```

Comment:
It might be good to have a macro for that, so the `/ 2` could be documented.

Reply:
p, q, dp, dq, qp are half the size of the keypair size (defined by n, d). I will define a macro if you want.